### PR TITLE
[OSDEV-1004] Update the SQL for gathering data and modify the facility schema to follow the target specification.

### PR DESCRIPTION
**Pull request by vladsha-dev**
[[OSDEV-1004](https://opensupplyhub.atlassian.net/browse/OSDEV-1004)] Update the SQL for gathering data and modify the facility schema to follow the target specification.

The following changes have been made to the Logstash and OpenSearch services:
- Prepared the SQL script to collect all the necessary data for the `v1/facilities` API endpoint according to [the new API specification](https://opensupplyhub.github.io/open-supply-hub-api-docs/). Agreed upon and established a prioritization scale for gathering data related to the name, address, sector, parent_company, product_type, facility_type, processing_type, number_of_workers and location fields as follows:
    - Data from the approved claim.
    - Promoted matches (considered as promoted facility list items).
    - The most recently contributed data.
    For the country field, the same prioritization scale has been utilized except for 'Data from the approved claims' because the claimant cannot update the country in any way.
- Introduced a new set of Ruby scripts to filter and reorganize the incoming data at the Logstash app level, avoiding complex database queries that could lead to high database load.
- Updated the `facilities` index template for OpenSearch to define how new fields within the facility documents are stored and indexed by OpenSearch.
- Set up the main Logstash pipeline to run every 15 minutes(agreed with stakeholders).
 
All changes have been made to meet [the API specification](https://opensupplyhub.github.io/open-supply-hub-api-docs/) requirements for `v1/facilities` API endpoint as closely as possible.

### CODE REVIEW

File: src/logstash/scripts/facilities/facility_type.rb [hunk 1]
```diff
@@ -1,16 +1,16 @@
 def filter(event)
-  # Get the facility_type_value (assuming it's a field in the event)
+  # Get the facility_type_value (assuming it's a field in the event).
```
Comment (mazursasha1990) [inline src/logstash/scripts/facilities/facility_type.rb:2]: Do we really need those changes here and in other places? I mean adding dots at the end of the comment line.

File: doc/release/RELEASE-NOTES.md [hunk 2]
```diff
@@ -3,6 +3,41 @@ All notable changes to this project will be documented in this file.
 
 This project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html). The format is based on the `RELEASE-NOTES-TEMPLATE.md` file.
 
+
+## Release 1.15.0
+
+## Introduction
+* Product name: Open Supply Hub
+* Release date: June 26, 2024
```
Comment (mazursasha1990) [inline doc/release/RELEASE-NOTES.md:11]: The release date is June 29, 2024. Please, change it.

**Review by mazursasha1990 — CHANGES_REQUESTED** [review]
Left some comments.

**Comment by roman-stolar** [comment]
In general LGTM, didn't see code changes issue directly. Waiting for final version if code changes (without Draft).

**Comment by vladsha-dev** [comment]
> In general LGTM, didn't see code changes issue directly. Waiting for final version if code changes (without Draft).

Thank you, Roman! I need to make some changes. I will let you know when it is ready.

File: src/logstash/scripts/facilities/facility_type.rb (hunk 1 above)
Comment (mazursasha1990) [inline src/logstash/scripts/facilities/facility_type.rb:2]: Do we really need those changes here and in other places? I mean adding dots at the end of the comment line.

**Review by mazursasha1990 — CHANGES_REQUESTED** [review]

**Comment by VadimKovalenkoSNF** [comment]
> Set up the main Logstash pipeline to run every 15 minutes(agreed with stakeholders).

Could you explain the intention of this?

**Review by VadimKovalenkoSNF — CHANGES_REQUESTED** [review]
Ask for minor changes

**Comment by vladsha-dev** [comment]
> > Set up the main Logstash pipeline to run every 15 minutes(agreed with stakeholders).
> 
> Could you explain the intention of this?

To avoid overloading the database with frequent requests, we decided to set a 15-minute gap between each refresh of OpenSearch. Based on Mariel's explanation, this will be sufficient for now since there is no need for facilities to appear in the search in real time.

**Review by roman-stolar — APPROVED** [review]
LGTM, I think we need @protsack-stephan review here.

**Review by Innavin369 — APPROVED** [review]

**Pull request merged**